tools/tasks/task.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Base class for all tasks.
+ */
+export abstract class Task {
```

Comment:
You could provide another abstract class which uses a regular expression to test file endings. This way a subclass would only need to pass an array of file endings to this abstract class instead of overriding` precondition` (`shallRun` respectively, see below)